deeplake/core/vectorstore/deep_memory.py

```diff
@@ -0,0 +1,348 @@
+import uuid
+from typing import Any, Dict, Optional, List, Union, Callable, Tuple
+from time import time
+
+import numpy as np
+
+import deeplake
+from deeplake.enterprise.dataloader import indra_available
+from deeplake.constants import DEFAULT_QUERIES_VECTORSTORE_TENSORS
+from deeplake.core.dataset import Dataset
+from deeplake.core.vectorstore.deeplake_vectorstore import VectorStore
+from deeplake.client.client import DeepMemoryBackendClient
+
+
+class DeepMemory:
+    def __init__(
+        self,
+        dataset: Dataset,
+        embedding_function: Optional[Any] = None,
+        token: Optional[str] = None,
+    ):
+        """Based Deep Memory class to train and evaluate models on DeepMemory managed service.
+
+        Args:
+            dataset (Dataset): deeplake dataset object.
+            embedding_function (Optional[Any], optional): Embedding funtion class used to convert queries/documents to embeddings. Defaults to None.
+            token (Optional[str], optional): API token for the DeepMemory managed service. Defaults to None.
+
+        Raises:
+            ImportError: if indra is not installed
+        """
+        self.dataset = dataset
+        self.token = token
+        self.embedding_function = embedding_function
+        self.client = DeepMemoryBackendClient(token=token)
+        self.queries_dataset = deeplake.dataset(
+            self.dataset.path + "_queries",
+            token=token,
+            read_only=False,
+        )
+        if len(self.queries_dataset) == 0:
+            self.queries_dataset.commit(allow_empty=True)
+
+    def train(
+        self,
+        queries: List[str],
+        relevance: List[List[Tuple[str, int]]],
+        embedding_function: Optional[Callable[[str], np.ndarray]] = None,
+        token: Optional[str] = None,
+    ):
+        """Train a model on DeepMemory managed service.
+
+        Args:
+            queries (List[str]): List of queries to train the model on.
+            relevance (List[List[Tuple[str, int]]]): List of relevant documents for each query.
+            embedding_function (Optional[Callable[[str], np.ndarray]], optional): Embedding funtion used to convert queries to embeddings. Defaults to None.
+            token (str, optional): API token for the DeepMemory managed service. Defaults to None.
+
+        Returns:
+            str: job_id of the training job.
+
```

Comment:
really need to give an example here how to provide queries and relevance as full code